application_sandbox/sample_application_framework/sample_application.h

```diff
@@ -27,7 +27,7 @@ namespace sample_application {
 
 const static VkSampleCountFlagBits kVkMultiSampledSampleCount =
     VK_SAMPLE_COUNT_4_BIT;
-const static VkFormat kDepthFormat = VK_FORMAT_D16_UNORM;
+const static VkFormat kDepthFormat = VK_FORMAT_D32_SFLOAT;
```

Comment:
We can't switch to D32_SFLOAT by default, as it is not guaranteed to be available on every Vulkan implementation.
The only format that we can actually guarantee is usable as a depth-stencil is D16_UNORM.

Reply:
If we need a higher resolution format, we can add a flag to the Sample Options use_high_quality_depth_format, that selects the highest quality depth format available based on hardware considerations.
We already do this if you want a stencil.

Reply:
with VK_FORMAT_D16_UNORM I am unable to produce z-fighting. I even tried to adjust the front and back plane distance but it doesn't really produce z-fighting.

So to Sample Options if use_high_quality_depth_format is true, then use VK_FORMAT_D32_SFLOAT?

Reply:
See: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap43.html#features-required-format-support for required formats.

Basically we should dynamic choose based on what is supported by the hardware
```
VK_FORMAT_FEATURE_DEPTH_STENCIL_ATTACHMENT_BIT feature must be supported for at least one of VK_FORMAT_X8_D24_UNORM_PACK32 and VK_FORMAT_D32_SFLOAT, and must be supported for at least one of VK_FORMAT_D24_UNORM_S8_UINT and VK_FORMAT_D32_SFLOAT_S8_UINT.
```

We will either have to be D32_SFLOAT or D24_UNORM_PACK32.
You can see how we do it with `options_.enable_stencil` sample_application.h:265
We will just have to create another helper like GetSupportedDepthStencilFormat (maybe GetSupportedHighQualityDepthFormat) if we want a higher precision format.